client/functions.lua

```diff
@@ -13,7 +13,18 @@ function QBCore.Functions.GetCoords(entity)
 end
 
 function QBCore.Functions.HasItem(items, amount)
```

Comment:
Could you write a header for the different overloads of this function? https://github.com/sumneko/lua-language-server/wiki/Annotations#overload

Reply:
Yessir. I amended the commit - let me know if I need to do anything differently workflow wise once a change has been requested. I see it in the list of commits for this PR.

Reply:
Does putting a space after the comment work? I haven't seen that before.

You can put types after the name Ex: @param items string|string[]

Should amount be optional, defaulting to any?

Unclear if items means has to have all the items, or items is if has any of the items.

Reply:
The space before param doesn't matter, we just don't do it because it comes without a space on auto-fill

Reply:
Also the params now aren't assigned a type, items should be a table and amount should be a number, none optional anymore with this usage but before it was optional

Reply:
The amount is required. I can make that clearer in the annotations - as well as refactor to basically say if `amount` is nil, then amount = 1. 

Not sure that makes it better per se, but rather enables lazy developers.

Reply:
Although, reading through the annotation documentation, a param is only required if it does not have a ? before it. So technically everything looks correct.

Reply:
Besides the type, it's missing that
`param paramName type`
`param amount number`

Reply:
> string|string[]

Amended commit.

Reply:
I still don't see the types?

Reply:
Sorry. Should be there now. When I go to amend a commit, it forces me to pull, not sure why because the tip of this branch has not changed. Anyway.

Reply:
It's good now yeah

Reply:
Can you clarify within the documentation that all items are required. I would actually suggest just creating a HasItems function instead of overloading this one, but it's fine.

Reply:
Can you also clarify within the documentation that amount is true if player has at least that amount and not player has to have exactly that amount.

Reply:
@Manason - I have made the requested changes. I did a bit of 'market research' and determined that most people that use this function are checking if it has 1 of the items - so I concluded that it would be best if we allowed the amount to be nil and to default to 1.

Furthermore, the functions name is self-describing - so this **should** be the behavior. The functions name is "HasItem" - so passing along just the item should return a boolean either way.

I have updated the annotations to reflect this - as well as marking amount as optional within annotations.